proxy/src/main/java/com/velocitypowered/proxy/VelocityServer.java

```diff
@@ -578,6 +579,22 @@ public void shutdown(boolean explicitExit, Component reason) {
       // done first to refuse new connections
       cm.shutdown();
 
+      int shutdownTimeout = 10;
+
+      try {
+        eventManager.fire(new ProxyPreShutdownEvent())
+                .toCompletableFuture()
+                .get(shutdownTimeout, TimeUnit.SECONDS);
+      } catch (TimeoutException ignored) {
+        logger.warn("Your plugins took over {} seconds to shut down while disconnecting players.",
```

Comment:
Ig something like this?
```suggestion
        logger.warn("Your plugins took over {} seconds during pre shut down.",
```